src/daemon/client.py

```diff
@@ -544,3 +544,5 @@ def send_message(
             message_as_bytes += constants.IRC_TERMINATION_DELIMITER
 
         self._key.data.out_buffer += message_as_bytes
+        # Set event loop selector to listen for write readiness on socket
+        self._key.data.toggle_writable(self._key.data, True)
```

Comment:
I understand that we expect the client to write to socket after we send a message to the client.

When we broadcast a message to multiple users on a channel, we are calling each of their send_message functions, thus modifying their respective selectors to EVENT_WRITE.

Each of those selectors will only be toggled to EVENT_READ once we receive a message from the respective clients.

Is my interpretation correct?

Reply:
Good question, we will toggle the selector to listen for EVENT_WRITE on each socket yet. They will be toggled back to EVENT_READ once we are done writing the message onto the socket